disco/bot/bot.py

```diff
@@ -326,11 +346,18 @@ def get_commands_for_message(self, require_mention, mention_rules, prefix, msg):
                     content = content.replace('<@{}>'.format(role), '', 1)
 
             content = content.lstrip()
-
-        if prefix and not content.startswith(prefix):
-            return []
         else:
-            content = content[len(prefix):]
+            # Scan through the prefixes to find the first one that matches.
```

Comment:
This block doesn't logically make sense to be a else branch of the prior if, they are unrelated.